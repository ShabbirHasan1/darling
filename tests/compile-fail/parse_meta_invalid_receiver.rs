extern crate proc_macro;
use darling::parse_meta;
use proc_macro::TokenStream;

struct NotMacroArgs;

pub fn invalid_receiver_type(args: TokenStream, _input: TokenStream) -> TokenStream {
    let args = parse_meta!(args as NotMacroArgs);

    unimplemented!()
}

fn main() {}
