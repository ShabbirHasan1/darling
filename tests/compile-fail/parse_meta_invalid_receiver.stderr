error[E0277]: the trait bound `NotMacroArgs: FromMeta` is not satisfied
 --> tests/compile-fail/parse_meta_invalid_receiver.rs:8:16
  |
8 |     let args = parse_meta!(args as NotMacroArgs);
  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromMeta` is not implemented for `NotMacroArgs`
  |
  = help: the following other types implement trait `FromMeta`:
            ()
            Arc<T>
            AtomicBool
            ExprArray
            ExprPath
            Flag
            HashMap<proc_macro2::Ident, V, S>
            HashMap<std::string::String, V, S>
          and $N others
  = note: this error originates in the macro `parse_meta` (in Nightly builds, run with -Z macro-backtrace for more info)
